### PR TITLE
Added axes node validation to DFTs operations

### PR DESCRIPTION
**Pull request by mhnap-lohika**
- *Fix DFTs axes node validation.*
 - *Add DFTs type prop tests for invalid nodes.*

### CODE REVIEW

**Comment by wilson-seok** [comment]
@mhnap-lohika Could you please rebase the PR? Several build checks are sill failed after multiple re-build. Rebase would solve it.

**Review by yeonbok — APPROVED** [review]

**Comment by yeonbok** [comment]
@artkuli  Hi Seems that you are maintainer of ngraph. Could you please take a look on this change? It is adding some validations for DFTs.

File: src/core/shape_inference/include/fft_base_shape_inference.hpp [hunk 1]
```diff
@@ -53,14 +53,23 @@ void shape_infer(const ov::op::util::FFTBase* op,
         // [n_0, ..., n_{r - 1}, 2] is interpreted as a complex tensor with the shape
         // [n_0, ..., n_{r - 1}].
         if (axes_shape.rank().is_static() && axes_are_known) {
+            ov::AxisSet axes_set;
             for (int64_t& axis : axes) {
+                NODE_VALIDATION_CHECK(op,
+                                      axis < static_cast<int64_t>(input_rank),
```
Comment (tomdol) [inline src/core/shape_inference/include/fft_base_shape_inference.hpp:59]: I'd rather perform the whole range validation in this place. This way the error message can contain information about the bounds in which the axis value must fit. The two separate messages below indicate that the axis value must implicitly be positive which contradicts with the docs.
```suggestion
-static_cast<int64_t>(input_rank) <= axis && axis < static_cast<int64_t>(input_rank),
```

File: src/core/shape_inference/include/rfft_common_validation.hpp [hunk 2]
```diff
@@ -80,9 +80,21 @@ void validate_axes(const ov::op::util::FFTBase* op,
     const int64_t axis_correction = (rfft_kind == RFFTKind::Forward) ? input_rank : (input_rank - 1);
     ov::AxisSet axes_set;
     for (int64_t& axis : axes) {
+        NODE_VALIDATION_CHECK(op,
```
Comment (tomdol) [inline src/core/shape_inference/include/rfft_common_validation.hpp:83]: The same suggestion as above.

**Review by tomdol — CHANGES_REQUESTED** [review]

File: src/core/shape_inference/include/fft_base_shape_inference.hpp [hunk 3]
```diff
@@ -53,14 +53,23 @@ void shape_infer(const ov::op::util::FFTBase* op,
         // [n_0, ..., n_{r - 1}, 2] is interpreted as a complex tensor with the shape
         // [n_0, ..., n_{r - 1}].
         if (axes_shape.rank().is_static() && axes_are_known) {
+            ov::AxisSet axes_set;
             for (int64_t& axis : axes) {
+                NODE_VALIDATION_CHECK(op,
```
Comment (mbencer) [inline src/core/shape_inference/include/fft_base_shape_inference.hpp:58]: Due to complex input represented like {D0, D1, D2, .., 2} (the last dim represents real and imaginary part), axes should be in range  `axis < input_rank-1`

File: src/core/shape_inference/include/rfft_common_validation.hpp [hunk 4]
```diff
@@ -80,9 +80,21 @@ void validate_axes(const ov::op::util::FFTBase* op,
...
+                              axis < static_cast<int64_t>(input_rank),
```
Comment (mbencer) [inline src/core/shape_inference/include/rfft_common_validation.hpp:84]: the same here

**Review by mbencer — CHANGES_REQUESTED** [review]

File: src/core/shape_inference/include/fft_base_shape_inference.hpp (hunk 3 above)
Comment (mhnap-lohika) [reply src/core/shape_inference/include/fft_base_shape_inference.hpp:58]: Yeah, we have check for this few lines below - https://github.com/openvinotoolkit/openvino/blob/5177db82e0c4af70ca8b569388eb04455198ebf3/src/core/shape_inference/include/fft_base_shape_inference.hpp#L79
But we can merge them together.

File: src/core/shape_inference/include/fft_base_shape_inference.hpp (hunk 1 above)
Comment (mhnap-lohika) [reply src/core/shape_inference/include/fft_base_shape_inference.hpp]: done

File: src/core/shape_inference/include/rfft_common_validation.hpp (hunk 2 above)
Comment (mhnap-lohika) [reply src/core/shape_inference/include/rfft_common_validation.hpp:92]: done

File: src/core/shape_inference/include/fft_base_shape_inference.hpp (hunk 3 above)
Comment (mhnap-lohika) [reply src/core/shape_inference/include/fft_base_shape_inference.hpp:60]: done

File: src/core/shape_inference/include/rfft_common_validation.hpp (hunk 4 above)
Comment (mhnap-lohika) [reply src/core/shape_inference/include/rfft_common_validation.hpp]: done

**Review by mbencer — APPROVED** [review]

**Review by tomdol — APPROVED** [review]

**Review by wilson-seok — APPROVED** [review]

**Pull request merged**